internal/provisioner/kops_provisioner_installation_sli.go

```diff
@@ -0,0 +1,130 @@
+				{
+					Name:        "requests-availability",
+					Objective:   99.9,
+					Description: "Availability metric for mattermost API",
+					SLI: slothv1.SLI{Events: &slothv1.SLIEvents{
+						ErrorQuery: "sum(rate(mattermost_api_time_count{job='mm-" + clusterInstallation.Namespace[0:4] + "',code=~'(5..|429)'}[{{.window}}]))",
+						TotalQuery: "sum(rate(mattermost_api_time_count{job='mm-" + clusterInstallation.Namespace[0:4] + "'}[{{.window}}]))",
+					}},
+					Alerting: slothv1.Alerting{
+						Name: "mm-" + clusterInstallation.Namespace[0:4] + "-service-HighAPIErrorRate",
+						Labels: map[string]string{
+							"category": "availability",
+						},
+						Annotations: map[string]string{
+							"summary": "High error rate on requests responses",
+						},
+					},
+				}},
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+	defer cancel()
+	_, err := k8sClient.SlothClientsetV1.SlothV1().PrometheusServiceLevels("prometheus").Create(ctx, sli, metav1.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "failed to create cluster installation sli")
+	}
+	return nil
+}
+
+func (provisioner *KopsProvisioner) createIfNotExistInstallationSLI(clusterInstallation *model.ClusterInstallation, k8sClient *k8s.KubeClient) error {
+	wait := 60
+	sli := &slothv1.PrometheusServiceLevel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterInstallation.InstallationID,
+			Labels: map[string]string{
+				"app":     "kube-prometheus-stack",
+				"release": "prometheus-operator",
+			},
+		},
+		Spec: slothv1.PrometheusServiceLevelSpec{
+			Service: clusterInstallation.InstallationID,
+			Labels: map[string]string{
+				"owner": "sreteam",
+			},
+			SLOs: []slothv1.SLO{
+				{
+					Name:        "requests-availability",
+					Objective:   99.9,
+					Description: "blah blah",
+					SLI: slothv1.SLI{Events: &slothv1.SLIEvents{
+						ErrorQuery: "sum(rate(mattermost_api_time_count{job='mm-" + clusterInstallation.Namespace[0:4] + "',code=~'(5..|429)'}[{{.window}}]))",
+						TotalQuery: "sum(rate(mattermost_api_time_count{job='mm-" + clusterInstallation.Namespace[0:4] + "'}[{{.window}}]))",
+					}},
+					Alerting: slothv1.Alerting{
+						Name: "mm-" + clusterInstallation.Namespace[0:4] + "-service-HighAPIErrorRate",
+						Labels: map[string]string{
+							"category": "availability",
+						},
+						Annotations: map[string]string{
+							"summary": "High error rate on requests responses",
+						},
+					},
+				}},
+		},
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(wait)*time.Second)
+	defer cancel()
+	_, err := k8sClient.SlothClientsetV1.SlothV1().PrometheusServiceLevels("prometheus").Get(ctx, sli.GetName(), metav1.GetOptions{})
+	if err != nil && !k8sErrors.IsNotFound(err) {
+		return err
+	}
+
+	if err != nil && k8sErrors.IsNotFound(err) {
+		k8sClient.SlothClientsetV1.SlothV1().PrometheusServiceLevels("prometheus").Create(ctx, sli, metav1.CreateOptions{})
```

Comment:
Instead of redefining the sloth file in this function, can't we just call `createInstallationSLI` above in the case that the file isn't found?